### PR TITLE
[16.0][ADD] delivery_carrier_manual_weight

### PR DESCRIPTION
**Pull request by JordiMForgeFlow**

### CODE REVIEW

**Review by PauBForgeFlow — APPROVED** [review]
Functional review! LGTM!

**Review by AaronHForgeFlow — APPROVED** [review]
Code review LGTM

**Comment by JordiMForgeFlow** [comment]
Hi @rousseldenis, could we get a review and merge on this one? :)

File: delivery_carrier_manual_weight/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    is_manual_weight = fields.Boolean(related="carrier_id.is_manual_weight")
+    weight_manual = fields.Float(string="Weight (Manual)", digits="Stock Weight")
+    shipping_weight_manual = fields.Float(string="Weight for Shipping (Manual)")
+
+    @api.depends(
+        "move_ids", "carrier_id", "carrier_id.is_manual_weight", "weight_manual"
```
Comment (rousseldenis) [inline delivery_carrier_manual_weight/models/stock_picking.py]: You cannot do that as if configuration changes, every picking will be recomputed.

```suggestion
        "move_ids", "carrier_id", "weight_manual"
```

File: delivery_carrier_manual_weight/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        for rec in manual_weight:
+            rec.weight = rec.weight_manual
+        return super(StockPicking, self - manual_weight)._cal_weight()
+
+    @api.depends(
+        "move_line_ids.result_package_id",
+        "move_line_ids.result_package_id.shipping_weight",
+        "weight_bulk",
+        "carrier_id",
+        "carrier_id.is_manual_weight",
```
Comment (rousseldenis) [inline delivery_carrier_manual_weight/models/stock_picking.py]: Same as above

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Looks great.

One important change

**Comment by JordiMForgeFlow** [comment]
totally agree @rousseldenis , thank you! :) It is changed now

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**